### PR TITLE
Little refactor, remove nonstandard js

### PR DESCRIPTION
**Pull request by callumacrae**

### CODE REVIEW

File: src/js/steps/letters/generateOverlay.js [hunk 1]
```diff
@@ -168,16 +168,16 @@ module.exports = function (options, $events) {
...
-              letter.characters
-                .map(character => ({ ...character, pageIndex: letter.index }))
-                .filter(character => character.character === letter.default_character)
-            );
+          .reduce(function (arr, letter) {
+            var characters = letter.characters
+              .map(character => $.merge(character, { pageIndex: letter.index }))
+              .filter(character => character.character === letter.default_character);
+
+            return [...arr, ...characters];
```
Comment (djgrant) [inline src/js/steps/letters/generateOverlay.js]: Still non-standard.

File: src/js/steps/letters/generateOverlay.js [hunk 2]
```diff
@@ -168,16 +168,16 @@ module.exports = function (options, $events) {
        */
       function revertCharsToOriginal(callback) {
         data.combinedLetters
-          .map((letter, index) => ({ ...letter, index }))
+          .map((letter, index) => $.merge(letter, { index }))
```
Comment (djgrant) [inline src/js/steps/letters/generateOverlay.js]: $.merge will do something different to {...letter, index}.

File: src/js/steps/letters/generateOverlay.js (hunk 1 above)
Comment (callumacrae) [inline src/js/steps/letters/generateOverlay.js]: It's straight from the documentation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator

File: src/js/steps/letters/generateOverlay.js (hunk 2 above)
Comment (callumacrae) [inline src/js/steps/letters/generateOverlay.js]: How so? what does `{...letter, index}` do?

File: src/js/steps/letters/generateOverlay.js (hunk 1 above)
Comment (djgrant) [inline src/js/steps/letters/generateOverlay.js]: sorry, array. I was still waking up.

File: src/js/steps/letters/generateOverlay.js (hunk 2 above)
Comment (djgrant) [inline src/js/steps/letters/generateOverlay.js]: well, isn't $.merge for smushing together 2 arrays and removing duplicate values?

`{...letter, index}` creates a new object with a deep copy of `letter` and another property `index` with the literal key `index` and the object `index` as the actual value.

Comment (callumacrae) [inline src/js/steps/letters/generateOverlay.js]: `$.merge` is the equivalent of `_.assign` or `Object.assign` (actually, I could have just used `Object.assign`.

The code above is pretty much what you're describing, except it didn't make a copy. Does it have to be a copy? It's easy to change if it does.

```js
$.merge(letter, { index });

// ->

$.merge(letter, { index: index });

// ->

letter.index = index; return letter
```

Comment (djgrant) [inline src/js/steps/letters/generateOverlay.js]: Nice, thanks for the explanation. Pretty sure that will be fine.

**Comment by callumacrae** [comment]
Cool, so good to merge?

**Comment by djgrant** [comment]
I haven't tested but happy with code.

**Pull request merged**